pkg/controller/keycloakclient/keycloakclient_reconciler.go

```diff
@@ -122,6 +124,68 @@ func roleMatches(a kc.RoleRepresentation, b kc.RoleRepresentation) bool {
 	return a.Name == b.Name
 }
 
+func (i *KeycloakClientReconciler) ReconcileScopeMappings(state *common.ClientState, cr *kc.KeycloakClient, desired *common.DesiredClusterState) {
+	if cr.Spec.ScopeMappings == nil {
+		cr.Spec.ScopeMappings = &kc.MappingsRepresentation{}
+	}
+	for clientID, clientMappings := range cr.Spec.ScopeMappings.ClientMappings {
+		clientMappings.Client = clientID
+	}
+
+	mappingsNew := scopeMappingDifference(cr.Spec.ScopeMappings, state.ScopeMappings)
+	if mappingsNew.RealmMappings != nil {
+		desired.AddAction(i.getCreatedClientRealmScopeMappingsState(state, cr, &mappingsNew.RealmMappings))
+	}
+	for _, clientMappings := range mappingsNew.ClientMappings {
+		desired.AddAction(i.getCreatedClientClientScopeMappingsState(state, cr, clientMappings.DeepCopy()))
+	}
+
+	mappingsDeleted := scopeMappingDifference(state.ScopeMappings, cr.Spec.ScopeMappings)
+	if mappingsDeleted.RealmMappings != nil {
+		desired.AddAction(i.getDeletedClientRealmScopeMappingsState(state, cr, &mappingsDeleted.RealmMappings))
+	}
+	for _, clientMappings := range mappingsDeleted.ClientMappings {
+		desired.AddAction(i.getDeletedClientClientScopeMappingsState(state, cr, clientMappings.DeepCopy()))
+	}
+}
+
+func scopeMappingDifference(a *kc.MappingsRepresentation, b *kc.MappingsRepresentation) (d *kc.MappingsRepresentation) {
```

Comment:
The logic in this function is not very straightforward. Some function comment would be useful here.

Reply:
done

Reply:
Thanks!